File: zeta/py/completed/1149.py
N = int(input())
C = [list(map(int, input().split())) for _ in range(N)]

T = [[0, 0, 0] for i in range(N)]
T[0] = C[0][:]

for i in range(1, N):
    T[i][0] = min((T[i - 1][1], T[i - 1][2])) + C[i][0]
    T[i][1] = min((T[i - 1][0], T[i - 1][2])) + C[i][1]
    T[i][2] = min((T[i - 1][0], T[i - 1][1])) + C[i][2]

print(min((T[N-1][0], T[N-1][1], T[N-1][2])))

File: storage/zeta/kt/completed/1149.kt
import kotlin.math.min

fun List<Int>.toTriple(): Triple<Int, Int, Int> {
    return Triple(this[0], this[1], this[2])
}


fun solve(N: Int, C: Array<Triple<Int, Int, Int>>): Int {
    val T = Array<Triple<Int, Int, Int>>(N) { Triple(0, 0, 0) }
    T[0] = C[0]

    for (i in 1 until N) {
        T[i] = Triple(
            min(T[i - 1].second, T[i - 1].third) + C[i].first,
            min(T[i - 1].first, T[i - 1].third) + C[i].second,
            min(T[i - 1].first, T[i - 1].second) + C[i].third
        )
    }

    return minOf(T[N - 1].first, T[N - 1].second, T[N - 1].third)
}

fun main() {
    val N = readln().toInt()
    val C = Array<Triple<Int, Int, Int>>(N) {
        readln().split(' ').map { s: String ->
            s.toInt()
        }.toTriple()
    }

    println(solve(N, C))
}